src/Microsoft.SqlTools.ServiceLayer/Migration/MigrationService.cs

```diff
@@ -186,203 +183,134 @@ public bool TryGetData(string column, out object value)
-                bool returnValue = LogonUser(
-                    userName, 
-                    domainName, 
-                    parameters.Password,
-                    LOGON32_LOGON_NEW_CREDENTIALS,
-                    LOGON32_PROVIDER_WINNT50,
-                    out safeAccessTokenHandle);
-
-                if (!returnValue)
+                return new ErrorModel()
                 {
-                    int ret = Marshal.GetLastWin32Error();
-                    string errorMessage = new Win32Exception(Marshal.GetLastWin32Error()).Message;
-                    await requestContext.SendError(errorMessage);
-                    return;
-                }
-                await WindowsIdentity.RunImpersonated(
-                safeAccessTokenHandle,
-                // User action  
-                async () =>
+                    ErrorId = e.ErrorID.ToString(),
+                    Message = e.Message,
+                    ErrorSummary = e.ErrorSummary,
+                    PossibleCauses = e.PossibleCauses,
+                    Guidance = e.Guidance,
+                };
+            }).ToList();
+        }
+        internal List<MigrationAssessmentInfo> ParseAssessmentResult(IList<ISqlMigrationAssessmentResult> assessmentResults, Dictionary<string, ISqlMigrationAssessmentResult> assessmentResultLookup)
+        {
+            return assessmentResults.Select(r =>
+            {
+                var check = assessmentResultLookup[CreateAssessmentResultKey(r)].Check;
+                return new MigrationAssessmentInfo()
                 {
-                    if(!Directory.Exists(parameters.Path)){
-                        await requestContext.SendError("Cannot connect to file share");
-                    } else {
-                        await requestContext.SendResult(true);
-                    }
-                }
-                );
-            } 
-            else 
+                    CheckId = check.Id,
+                    Description = check.Description,
+                    DisplayName = check.DisplayName,
+                    HelpLink = check.HelpLink,
+                    Level = check.Level.ToString(),
+                    TargetName = r.AppliesToMigrationTargetPlatform.ToString(),
+                    DatabaseName = r.DatabaseName,
+                    ServerName = r.ServerName,
+                    Tags = check.Tags.ToArray(),
+                    RulesetName = Engine.Configuration.DefaultRuleset.Name,
+                    RulesetVersion = Engine.Configuration.DefaultRuleset.Version.ToString(),
+                    RuleId = r.FeatureId.ToString(),
+                    Message = r.Message,
+                    AppliesToMigrationTargetPlatform = r.AppliesToMigrationTargetPlatform.ToString(),
+                    IssueCategory = r.IssueCategory.ToString(),
+                    ImpactedObjects = ParseImpactedObjects(r.ImpactedObjects).ToArray()
+                };
+            }).ToList();
```

Comment:
it looks we do a ToList here then immediately do a ToArray on the return value.  It maybe better to use ToArray here directly to avoid so much conversation & data copying.